### PR TITLE
Search directory elements elk

### PR DESCRIPTION
**Pull request by ghazwarhili**

### CODE REVIEW

File: src/main/java/org/gridsuite/directory/server/DirectoryService.java [hunk 1]
```diff
@@ -439,35 +446,31 @@ private void deleteSubElements(UUID elementUuid, String userId) {
...
+        path.addAll(ancestors.stream().map(ElementAttributes::toElementAttributes).toList());
 
         return path;
     }
 
+    private DirectoryElementEntity getParentElement(@NotNull DirectoryElementEntity currentElement) {
+        return repositoryService.getElementEntity(currentElement.getParentId())
+                .orElseThrow(() -> DirectoryException.createElementNotFound(ELEMENT, currentElement.getParentId()));
+    }
+
```
Comment (YenguiSeddik) [inline src/main/java/org/gridsuite/directory/server/DirectoryService.java:473]: we already have a method: private ElementAttributes getParentElement(UUID elementUuid)

File: src/main/java/org/gridsuite/directory/server/dto/elasticsearch/DirectoryElementInfos.java [hunk 2]
```diff
@@ -56,4 +58,11 @@ public class DirectoryElementInfos {
 
     @Field(type = FieldType.Boolean)
     private Boolean isPrivate;
+
+    @Transient
+    private List<String> directoryName;
```
Comment (YenguiSeddik) [inline src/main/java/org/gridsuite/directory/server/dto/elasticsearch/DirectoryElementInfos.java:63]: maybe directoryPathName

File: src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+ * @author Ghazwa Rehili <ghazwa.rehili at rte-france.com>
+ */
+@Service
+public class DirectoryElementInfosService {
+
+    private static final int PAGE_MAX_SIZE = 10;
+
+    private final ElasticsearchOperations elasticsearchOperations;
+
+    static final String ELEMENT_NAME = "name";
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java:32]: name.fullascii

File: src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import lombok.NonNull;
+import java.util.List;
+
+/**
+ * @author Ghazwa Rehili <ghazwa.rehili at rte-france.com>
+ */
+@Service
+public class DirectoryElementInfosService {
+
+    private static final int PAGE_MAX_SIZE = 10;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java:28]: Why 10 ?

File: src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java (hunk 3 above)
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java:32]: Add consants for othser terms (isPrivate, owner, type)

File: src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        boolQueryBuilder.should(isPublicQuery._toQuery());
+        boolQueryBuilder.filter(elementSearchQuery._toQuery());
+        BoolQuery finalQuery = boolQueryBuilder.build();
+
+        NativeQuery nativeQuery = new NativeQueryBuilder()
+                .withQuery(finalQuery._toQuery())
+                .withPageable(PageRequest.of(0, PAGE_MAX_SIZE))
+                .build();
+
+        return Lists.newArrayList(elasticsearchOperations.search(nativeQuery, DirectoryElementInfos.class)
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java:61]:
```suggestion
        return Lists.newArrayList(elasticsearchOperations.search(nativeQuery, DirectoryElementInfos.class)
```     return elasticsearchOperations.search(nativeQuery, DirectoryElementInfos.class)
                .stream()
                .map(SearchHit::getContent)
                .toList();

File: src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                .map(SearchHit::getContent));
+    }
+
+    public static String escapeLucene(String s) {
+        StringBuilder sb = new StringBuilder(s.length() + 16);
+
+        for (int i = 0; i < s.length(); ++i) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/directory/server/services/DirectoryElementInfosService.java:68]: Use the same characters as study server ?

**Review by jamal-khey — APPROVED** [review]
looks good to me , only small remark

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**